utils.py

```diff
@@ -2664,6 +2664,52 @@ def distRHPZ(G, Gd, RHP_Z):
 
     return Dist_RHPZ
 
+def ssr_solve(A, B, C, D):
+    """
+    Solves the zeros and poles of a state-space representation of a system.
+
+    :param A: System state matrix
+    :param B: matrix
+    :param C: matrix
+    :param D: matrix
+
+    For information on the meanings of A, B, C, and D consult Skogestad 4.1.1
+
+    :return: dict, with keys:
```

Comment:
What's the benefit of returning a `dict` vs just the two values you calculate?

Reply:
I have seen many modules returning dictionaries, I assume (because this is the effect I have felt as a user) that it makes the functions easier to use.

For example, scipy.optimize functions return an OptimizeResult object, which is derived from a dictionary. Whenever I've had to use this module I've appreciated being able to call result['success'] or result['nfev'] to determine if the algorithm reported convergence and how many function evaluations were used. I find it more 'pythonic' than having to index an arbitrary position in a list or tuple as it eliminates magic numbers in code.

I personally would much rather use functions where the docstring lists the dictionary keys for the return values I'm interested in rather than having to count through elements to find the value I'm interested in.

Obviously this is a bit redundant with only two return values, but I was leaving room for additional return values.

---

Let me know if you'd like this function changed to return a tuple.